modules/vf-graphql-holochain/queries/agent.ts

```diff
@@ -13,32 +13,50 @@ import { mapZomeFn, serializeHash, deserializeHash } from '../connection.js'
 import {
   AccountingScope,
   Agent,
+  AgentConnection,
+  AgentEdge,
   Organization,
-  Person
+  OrganizationConnection,
+  Person,
+  PersonConnection
 } from '@valueflows/vf-graphql'
 import { AgentPubKey } from '@holochain/client'
 import { AgentResponse } from '../mutations/agent'
+import { AgentSearchInput, PagingParams } from '../resolvers/zomeSearchInputTypes.js'
 
 export interface RegistrationQueryParams {
   pubKey: AgentPubKey,
 }
+export type AgentWithType = Agent & { agentType: string }
+export interface AgentWithTypeResponse {
+  agent: AgentWithType
+}
+export interface AgentEdgeWithTypeEdge extends Omit<AgentEdge, 'node'> {
+  node: AgentWithType
+}
+export interface AgentConnectionWithType extends Omit<AgentConnection, 'edges'> {
+  edges: AgentEdgeWithTypeEdge[]
+}
 
 export default (dnaConfig: DNAIdMappings, conductorUri: string) => {
 
   //assumes there is a link from agentPubKey to a Person entry, but what if link cannot be resolved?
   const readMyAgent = mapZomeFn<null, AgentResponse>(dnaConfig, conductorUri, 'agent', 'agent', 'get_my_agent')
-  const readAgent = mapZomeFn<ReadParams, AgentResponse>(dnaConfig, conductorUri, 'agent', 'agent', 'get_agent')
+  const readAgent = mapZomeFn<ReadParams, AgentWithTypeResponse>(dnaConfig, conductorUri, 'agent', 'agent', 'get_agent')
+  const readAll = mapZomeFn<PagingParams, AgentConnectionWithType>(dnaConfig, conductorUri, 'agent', 'agent_index', 'read_all_agents')
+  const readAllAgentType = mapZomeFn<AgentSearchInput, AgentConnection>(dnaConfig, conductorUri, 'agent', 'agent_index', 'query_agents')
 
   return {
     // :TODO: is myAgent always a 'Person' in Holochain, or will we allow users to act in an Organization context directly?
     myAgent: injectTypename('Person', async (root, args): Promise<Agent> => {
       return (await readMyAgent(null)).agent
     }),
-
-    // TODO: not totally sure on the significance of the `__typename` field injection, but once the `type` field is added to Agent, we could conditionally inject `Person` or `Organization`
-    agent: injectTypename('Person', async (root, args): Promise<Agent> => {
-      return (await readAgent({ address: args.id })).agent
-    }),
+    // NOTE: should we drop the `agentType` field before passing through to client?
```

Comment:
```suggestion
```
no need, the graphql resolvers will automatically not 
A. let it be queried for
B. let it pass through to the result